File: front-end/src/service/journalService.js
import axios from 'axios';

const API_URL = 'http://localhost:8083/journals';

function getAuthHeader() {
  const token = localStorage.getItem("token");
  return token ? { Authorization: `Bearer ${token}` } : {};
}

const JournalService = {
  getAllJournals: () => axios.get(API_URL, { headers: getAuthHeader() }),
  getJournalsByUser: (userId) => axios.get(`${API_URL}/user/${userId}`, { headers: getAuthHeader() }),
  addJournal: (journalData) => axios.post(API_URL, journalData, { headers: getAuthHeader() }),
  addToJournal: (data) => axios.post("http://localhost:8083/journals", data, { headers: getAuthHeader() }),
  deleteJournal: (journalId) => axios.delete(`${API_URL}/${journalId}`, { headers: getAuthHeader() }),
  updateJournal:(editingId, updatedJournal) => axios.put(`${API_URL}/${editingId}`, updatedJournal, { headers: getAuthHeader() })
};

export default JournalService;

File: JournalService/src/main/java/com/demo/journalservice/service/JournalService.java
package com.demo.journalservice.service;

import com.demo.journalservice.dto.JournalRequest;
import com.demo.journalservice.entity.Journal;

import java.util.List;

public interface JournalService {

    Journal addJournal(Journal journal);
    Journal addToJournal(JournalRequest request);

    List<Journal> getAllJournals();
    Journal getJournal(int journalId);
    List<Journal> getJournalsByUser(int userId);
    Journal updateJournal(int journalId, JournalRequest journal);
    String deleteJournal(int journalId);
}
